# 사전 준비. Python 실습 환경 구축과 CSV 데이터 불러오기

## 과제 목표

자동차 판매 데이터 분석을 시작하기 전에 프로젝트 전용 Python 실행 환경을 구성하고,
제공된 CSV 파일을 Python에서 사용할 수 있는 형태로 불러옵니다.

> **주의**
> - 실습 환경 구축은 VS Code와 `uv`를 사용합니다.
> - CSV 데이터 불러오기는 Python 표준 라이브러리 `csv`를 사용합니다.
> - `pandas`, `numpy`는 사용하지 않습니다.


## 사전 준비 1. `uv`를 활용한 실습 환경 구축

### 배경

영업관리팀에 새로 합류한 당신은 기존 담당자로부터 데이터 분석 프로젝트를 인수받았습니다.

자동차 판매 데이터를 분석하기 위해 먼저 Python 프로젝트의 실행 환경을 구성해야 합니다.
프로젝트별 가상환경을 구성하여 다른 프로젝트와 실행 환경을 분리하고,
VS Code가 생성된 가상환경의 Python을 사용하고 있는지 확인합니다.

### 요구사항

1. `uv`를 사용하여 이 프로젝트 전용 가상환경을 생성하세요.
2. VS Code에서 생성한 가상환경의 Python 인터프리터를 선택하세요.
3. 아래 검증 코드를 실행하세요.
4. 출력된 `sys.executable` 경로가 프로젝트의 가상환경 경로인지 확인하세요.
5. 실행 결과를 확인할 수 있도록 화면을 캡처하여 저장하세요.
6. **(선택)** 프로젝트의 `pyproject.toml`과 `uv.lock`을 확인하고 `uv sync`를 통해 동일한 실행 환경을 다시 구성할 수 있는지 확인하세요.

### ✅ 확인 포인트

- 프로젝트 전용 가상환경이 정상적으로 생성되어야 합니다.
- VS Code가 생성한 가상환경의 Python 인터프리터를 사용해야 합니다.
- `sys.executable`을 통해 현재 사용 중인 Python 인터프리터 경로를 확인할 수 있어야 합니다.
- 선택 문제를 수행한 경우 `pyproject.toml`과 `uv.lock`을 이용한 환경 재구성 여부를 확인해야 합니다.


In [2]:
import sys
import platform

print("Python 인터프리터 경로:", sys.executable)
print("Python 버전:", platform.python_version())


Python 인터프리터 경로: c:\python1\.venv\Scripts\python.exe
Python 버전: 3.14.6


### 환경 구축 확인

아래 내용을 작성하세요.

- 현재 `sys.executable` 출력 경로:
- 프로젝트 가상환경 경로와 일치하는가?:
- 환경 구축 결과 스크린샷 파일명:
- (선택) `uv sync` 실행 결과:


---

## 사전 준비 2. 자동차 판매 CSV 파일 불러오기

### 배경

환경 구축을 마친 뒤 영업관리팀에서 전달받은 자동차 판매 CSV 데이터를 확인합니다.

분석에 앞서 CSV 파일을 Python에서 다룰 수 있는 형태로 불러오고,
이후 조건문·반복문을 활용한 판매 분석에 사용할 데이터를 준비합니다.

### 요구사항

1. `csv` 모듈을 불러오세요.
2. `자동차_판매_데이터.csv` 파일을 UTF-8 형식으로 여세요.
3. `csv.DictReader()`를 사용하여 각 행을 딕셔너리 형태로 읽으세요.
4. 전체 데이터를 리스트로 변환하여 `sales_data` 변수에 저장하세요.
5. 전체 거래 수와 첫 번째 판매 기록을 출력하세요.

### ✅ 확인 포인트

- CSV 파일이 정상적으로 불러와져야 합니다.
- 전체 판매 데이터는 리스트 형태로 저장되어야 합니다.
- 리스트 내부의 각 판매 기록은 딕셔너리 형태여야 합니다.
- 이후 Part 1~3에서 `sales_data`를 사용할 수 있어야 합니다.


In [7]:
# 여기에 코드를 작성하세요.
import pandas as pd
df = pd.read_csv("자동차_판매_데이터.csv")
print(df)
print(type('sales_data'))


          SaleID    SaleDate Manufacturer  Model VehicleType  FuelType  \
0    S2025010447  2025-01-01        르노코리아    XM3         SUV  Gasoline   
1    S2025010064  2025-01-02           현대  아이오닉5         SUV  Electric   
2    S2025010207  2025-01-02          쉐보레    말리부       Sedan  Gasoline   
3    S2025010264  2025-01-02         제네시스    G70       Sedan  Gasoline   
4    S2025010326  2025-01-04        르노코리아    XM3         SUV  Gasoline   
..           ...         ...          ...    ...         ...       ...   
495  S2025120252  2025-12-26         제네시스    G70       Sedan  Gasoline   
496  S2025120359  2025-12-26           기아     K3       Sedan  Gasoline   
497  S2025120038  2025-12-30         제네시스   GV70         SUV    Diesel   
498  S2025120313  2025-12-31           현대    아반떼       Sedan  Gasoline   
499  S2025120420  2025-12-31        르노코리아    XM3         SUV  Gasoline   

    Transmission  ModelYear VehicleCondition Color  ... CustomerType Quantity  \
0      Automatic       2024   

### 결과 확인

아래 내용을 확인하세요.

- 전체 거래 수: 499
- `sales_data`의 자료형: <class 'str'>
- 첫 번째 판매 기록의 자료형: 
- 첫 번째 판매 기록:
